### ISSUE TITLE
Execution failed for task ':compileTestJava'.

### ISSUE THREAD

**Issue by tototto** [nus-tic4002-AY2021S2/forum#8]
Hi i have followed gradle the tutorial to set up gradle correctly. However i do not understand why they are not able to build junit test

> Task :compileJava UP-TO-DATE
> Task :processResources UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :compileTestJava FAILED
/Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/test/java/inputParser/inputParserTest.java:5: error: package org.junit does not exist
import org.junit.Test;
                ^
/Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/test/java/listHandler/listHandlerTest.java:2: error: cannot find symbol
import org.junit.Test;
                ^
  symbol:   class Test
  location: package org.junit
/Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/test/java/inputParser/inputParserTest.java:12: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class inputParserTest
/Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/test/java/inputParser/inputParserTest.java:22: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class inputParserTest
/Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/test/java/inputParser/inputParserTest.java:32: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class inputParserTest
/Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/test/java/listHandler/listHandlerTest.java:10: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class listHandlerTest
/Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/test/java/listHandler/listHandlerTest.java:19: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class listHandlerTest
/Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/test/java/listHandler/listHandlerTest.java:28: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class listHandlerTest
8 errors
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler error output for details.
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 7s

**Comment by tototto** [comment]
![image](https://user-images.githubusercontent.com/43071400/106376428-36487e00-63d0-11eb-9ba0-e5eb0d9d5b9c.png)

**Comment by tototto** [comment]
adding the dependency also does not work
![image](https://user-images.githubusercontent.com/43071400/106376445-66901c80-63d0-11eb-89c3-006c64258d9e.png)

**Comment by damithc** [comment]
Shouldn't it be `@Test`?

**Comment by tototto** [comment]
they do not recognise it
![image](https://user-images.githubusercontent.com/43071400/106377471-2f256e00-63d8-11eb-91d8-6f35060b0441.png)

**Comment by tototto** [comment]
and my Test file is already using 
![image](https://user-images.githubusercontent.com/43071400/106377490-4fedc380-63d8-11eb-8adc-963bd886df2e.png)

**Comment by damithc** [comment]
> and my Test file is already using
> ![image](https://user-images.githubusercontent.com/43071400/106377490-4fedc380-63d8-11eb-8adc-963bd886df2e.png)

Yes, I was referring to `@Test` in java code. What's the exact error message now, after adding the JUnit dependency to the gradle file?

**Comment by damithc** [comment]
BTW, your test class names are not PascalCase.

**Comment by tototto** [comment]
Testing started at 3:27 PM ...

> Task :compileJava
Note: /Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/main/java/duke/storage/ListHandler.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
> Task :processResources UP-TO-DATE
> Task :classes
> Task :compileTestJava FAILED
/Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/test/java/inputParser/inputParserTest.java:5: error: package org.junit does not exist
import org.junit.Test;
                ^
/Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/test/java/listHandler/listHandlerTest.java:2: error: cannot find symbol
import org.junit.Test;
                ^
  symbol:   class Test
  location: package org.junit
/Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/test/java/inputParser/inputParserTest.java:12: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class inputParserTest
/Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/test/java/inputParser/inputParserTest.java:22: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class inputParserTest
/Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/test/java/inputParser/inputParserTest.java:32: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class inputParserTest
/Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/test/java/listHandler/listHandlerTest.java:10: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class listHandlerTest
/Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/test/java/listHandler/listHandlerTest.java:19: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class listHandlerTest
/Users/tayjunwen/Desktop/TIC4001_Software_Engineering_Practicum/ip/src/test/java/listHandler/listHandlerTest.java:28: error: cannot find symbol
    @Test
     ^
  symbol:   class Test
  location: class listHandlerTest
8 errors
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler error output for details.
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 1s
3 actionable tasks: 2 executed, 1 up-to-date

**Comment by damithc** [comment]
See if refreshing (as noted in https://github.com/nus-cs2103-AY2021S2/forum/issues/74) helps. If not, push the exact code and give the link here so that someone can take a look.

**Comment by tototto** [comment]
if i add this in:

![image](https://user-images.githubusercontent.com/43071400/106377597-179ab500-63d9-11eb-9dee-8dea0272f02b.png)

then it is able to remove the build error but its deprecated

Testing started at 3:30 PM ...

> Task :compileJava UP-TO-DATE
> Task :processResources UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources NO-SOURCE
> Task :testClasses
> Task :test
Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.2/userguide/command_line_interface.html#sec:command_line_warnings
BUILD SUCCESSFUL in 1s
4 actionable tasks: 2 executed, 2 up-to-date
3:30:10 PM: Task execution finished 'test'.

**Comment by tototto** [comment]
i have pushed the code
https://github.com/tototto/ip

on my local i ran compileTestJava and Test using gradle

**Comment by tototto** [comment]
refreshing the gradle build file does not help

**Comment by Diwu-Yi** [comment]
> i have pushed the code
> https://github.com/tototto/ip
> 
> on my local i ran compileTestJava and Test using gradle

Hi, @tototto does this mean that you can successfully compile and run your code on your own machine but fail GitHub checks in your repo ?

If so, please check that you have pushed the passing version to your remote repo and merged into master (if your are doing this on a separate branch) because it really does not make sense that the code compiles on your machine but fails constantly on GitHub.

If the check just failed once and you are confident that it compiles, please click Re-run Job, sometimes this solve the problem (GitHub can be weird from time to time :)).

**Comment by tototto** [comment]
@Diwu-Yi 

I am not able to run the test on my machine and it also fail GitHub checks in my repo

**Comment by tohyuting** [comment]
Hi @tototto, I took a look at your code and I realised that the import statements you used for JUnit Test class is from `org.junit.Test` instead of `org.junit.jupiter.api.Test` as mentioned [here](https://se-education.org/guides/tutorials/junit.html). I have changed the import statements in both test cases codes (Sample screenshot below) and it is working on my end now. 
![image](https://user-images.githubusercontent.com/65058690/106379072-7ca7d800-63e4-11eb-9ec5-e847e2071178.png)

Do try and see if this solves your issue! :)
